src/libponyc/ast/token.h

```diff
@@ -147,6 +147,7 @@ typedef enum token_id
   TK_IF,
   TK_IFDEF,
   TK_IFTYPE,
+  TK_IFTYPE_CLAUSE,
```

Comment:
How would you feel about calling this `TK_IFTYPE_COND` instead of `TK_IFTYPE_CLAUSE`?

At least In the ponyc source code, we often use `clause` to refer to the body of a conditional, and `cond` to refer to the condition itself.  [Here's at least one example of this precedent](https://github.com/ponylang/ponyc/blob/75d9bbc0fc43413a5808f0eb91d5cded27d9279b/src/libponyc/ast/frame.c#L257-L258).